engine/src/main/java/org/terasology/rendering/nui/widgets/UIText.java

```diff
@@ -677,15 +677,15 @@ public void bindText(Binding<String> binding) {
      * contents of linesOfText or text accordingly.
      */
     public String getText() {
-        String arrayText = "";
-        if(linesOfText.length > 0) {
-            arrayText = String.join("\n", linesOfText);
-        }
-
-        if(text.get() != "") {
-            return text.get() + "\n" + arrayText;
+        if (linesOfText.length > 0) {
+            String arrayText = String.join("\n", linesOfText);
+            if (text.get().equals("")) {
+                return text.get() + "\n" + arrayText;
```

Comment:
In this case `text.get()` is the empty string `""` - why are we prepending a newline here? Should it rather be:

```java
 if (text.get().equals("")) {
    return arrayText;
} else {
    return text.get() + "\n" + arrayText;
}
```

Reply:
My mistake, it should be `!text.get().equals("")`. I will fix it.

Reply:
Should I just return `arrayText` instead of `text.get() + arrayText`? There was a discussion in original pr #3678 regarding this but resolved eventually.